cabal-install/Distribution/Client/CmdRun.hs

```diff
@@ -90,7 +121,8 @@ runAction (configFlags, configExFlags, installFlags, haddockFlags)
     baseCtx <- establishProjectBaseContext verbosity cliConfig
 
     targetSelectors <- either (reportTargetSelectorProblems verbosity) return
-                   =<< readTargetSelectors (localPackages baseCtx) targetStrings
+                   =<< readTargetSelectors (localPackages baseCtx)
+                         (take 1 targetStrings) -- we drop the exe's args
```

Comment:
Urk. Wouldn't it be better to split targetStrings first before doing anything with it? See (A).